### PR TITLE
Locale aware format smart

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Part of https://github.com/gnosis/gp-ui/issues/382

1. Fixed bug where `formatSmart` values < 0 would not respect locale

2. Fixed bug where values between 0.5* and 0.9* would be formatted to 1..5*
The `integerValue` BigNumber method rounds the integer by default :facepalm: 
Fixed by forcing it to round down `const integer = numberToConvert.integerValue(BigNumber.ROUND_DOWN)`

3. While at it, I added the option to make `formatSmart` NOT locale aware.
It was by default, but unlike `formatAmount`, there was no way to turn it off in case we wanted to.

4. Updated `husky` pre-commit hook lib, as it wouldn't run locally

5. And lastly, made sure the tests are capable of properly testing locale aware changes
This one took awhile...

6. All of that released as [`0.10.0-RC.0` ](https://www.npmjs.com/package/@gnosis.pm/dex-js/v/0.10.0-RC.0)

# Testing

Unit tests

Will add the RC version to `gp-ui` repo and test it there as well. -> https://github.com/gnosis/gp-ui/pull/417

### CODE REVIEW

File: src/const.ts [hunk 1]
```diff
@@ -34,6 +34,8 @@ export const DEFAULT_LARGE_NUMBER_PRECISION = 3
 export const DEFAULT_SMALL_LIMIT = '0.001'
 export const DEFAULT_THOUSANDS_SYMBOL = ','
 export const DEFAULT_DECIMALS_SYMBOL = '.'
+// Export symbols based on user locale
+export const { thousands: THOUSANDS_SYMBOL, decimals: DECIMALS_SYMBOL } = _getLocaleSymbols()
```
Comment (alfetopito) [inline src/const.ts:38]: Moved these to `const` so I can mock them on unit tests

File: src/utils/format.ts [hunk 2]
```diff
@@ -17,30 +17,26 @@ import {
...
-  }
-}
-
-const { thousands: THOUSANDS_SYMBOL, decimals: DECIMALS_SYMBOL } = _getLocaleSymbols()
-
-function _formatDecimalsForDisplay(numberToConvert: BigNumber) {
+function _formatDecimalsForDisplay(numberToConvert: BigNumber, decimalsSymbol: string) {
   // 2.00012366123.integerValue() ==> 2
-  const integer = numberToConvert.integerValue()
+  const integer = numberToConvert.integerValue(BigNumber.ROUND_DOWN)
```
Comment (alfetopito) [inline src/utils/format.ts:27]: Rounding bug making 0.5* to 0.9* be converted to 1..5... etc

File: src/utils/format.ts [hunk 3]
```diff
@@ -17,30 +17,26 @@ import {
...
   // 2.00012366123 <numberToConvert> - 2 <integer> ==> 00012366123
   const decimalsWithoutIntegerOrSymbol = numberToConvert.minus(integer).toString(10).slice(2)
   // 2 + <,|.> + 00012366123 ==> 2<,|.>00012366123
-  return integer + DEFAULT_DECIMALS_SYMBOL + decimalsWithoutIntegerOrSymbol
+  return integer + decimalsSymbol + decimalsWithoutIntegerOrSymbol
```
Comment (alfetopito) [inline src/utils/format.ts:31]: Make it locale aware/respect input parameter

**Comment by alfetopito** [comment]
Travis build is failing.
AFAICT, refreshing `yarn.lock` should fix that.
Doing that locally though is causing more issues with some lib update:
screenshot_2021-04-30_12-21-50

Tried some ways to fix that but got nowhere so far.

Still, the Travis failure does not block the release. I can run tests and all steps fine locally, so we can fix this later.

**Review by W3stside — APPROVED** [review]

**Review by anxolin — COMMENTED** [review]
APPROVE

**Pull request merged**